File: db/models/reservation.py
from db.database_init import get_db_instance
from sqlalchemy.orm import relationship


db = get_db_instance()

class Reservation(db.Model):
    __tablename__ = 'reservation'
    id = db.Column(db.Integer, primary_key=True)
    guest_count = db.Column(db.Integer)
    is_confirmed = db.Column(db.Boolean)
    is_canceled = db.Column(db.Boolean)
    time_and_date = db.Column(db.DateTime)
    customer_id = db.Column(db.Integer, db.ForeignKey("customer.id"))
    restaurant_id = db.Column(db.Integer, db.ForeignKey("restaurant.id"))

    def __repr__(self):
        return "<Reservation {}>".format(self.id)

File: web/src/types/reservation.ts
import { Restaurant } from "@/types/restaurant";
import { Customer } from "@/types/customer";

export interface Reservation {
    id: number;
    guest_count: number;
    is_confirmed: boolean;
    is_canceled: boolean;
    time_and_date: Date;
    customer: Customer;
    restaurant: Restaurant;
}
